src/base/async/int.c

```diff
@@ -1890,16 +1890,28 @@ static int redir_it(void)
     uint16_t lol_lo, lol_hi, sda_lo, sda_hi, sda_size, redver, mosver;
     uint8_t major, minor;
     int is_MOS;
+    char *fspec;
 
     /*
      * To start up the redirector we need
-     * (1) the list of list,
-     * (2) the DOS version and
-     * (3) the swappable data area.
+     * (1) the list of list
+     * (2) the DOS version
+     * (3) the swappable data area
      */
     if (HI(ax) != 0x3d)
 	return 0;
 
+    // Don't Run until the file is AUTOEXEC.BAT
+    fspec = SEG_ADR((char *), ds, dx);
+    if (!fspec)
+        return 0;
+    if (fspec[0] && fspec[1] == ':')
+        fspec += 2;
+    if (fspec[0] == '\\')
+        fspec++;
+    if (strcasecmp(fspec, "AUTOEXEC.BAT") != 0)
```

Comment:
autoexec.bat may not exist at all.

Reply:
Plus people use SHELL directive in config.sys
that alters the bat file to non-default one, because
the autoexec.bat is used to boot under the bare-metal.
I use exactly that trick, so this will not work.

Reply:
Yes I need to find out if DOS tests for its existence before trying to open. If it just tries to open it, the code is fine, but if it tests beforehand perhaps I need to discover some other method of knowing that config.sys has been processed.

Reply:
Ahh, I now see your SHELL comment. Have you any good ideas how to run this code after config.sys has been processed?

Reply:
But what is a purpose?
Is it only to get lastdrive from DOS
instead of from LOL?
Do these value really mismatch while
config.sys is being processed?

Reply:
Only purpose is to delay MFS initialisation until we can be sure that config.sys has been processed so that lastdrive value is fixed and CDS is in its final position. So yes getting lastdrive from DOS instead of LOL is goal.
Yep like I said with MS-DOS 6.22 lastdrive=Z in config.sys calling int21/0e from emufs.sys gave value of 5, but from delayed 32 bit init code I saw 26.
Incidentally there's a lastdrivehigh command in FreeDOS and MS-DOS 7.x that moves CDS into upper memory that we probably want to avoid happening after we've done the init (at least until I can kill the global CDS array access.)

Reply:
> Yep like I said with MS-DOS 6.22 lastdrive=Z in config.sys
> calling int21/0e from emufs.sys gave value of 5

But does the same value taken from LOL
is _not_ 5 at the same time?
What's wrong with CDSes if the drives we
need to redirect, are already defined at
that point?
By the way, what happens if we define
more drives in dosemu.conf than the default
lastdrive allows? Are they invisible till autoexec.bat?

Reply:
> But does the same value taken from LOL
is not 5 at the same time?

Quite possibly, but since the config.sys has lastdrive=Z it's not a lot of use being consistent but incorrect

> What's wrong with CDSes if the drives we
need to redirect, are already defined at
that point?

At the moment there's that global CDS array access that our LFN uses outside of the redirector, that's why I want to push some of the LFN truename inside the redirector as tweaked qualify_filename. If all the CDS magic happens inside the redirector, we can look up the CDS if necessary using int2f/1217 if we can't get it from the info passed in the call. At that point the CDS can move all it likes and we won't care, but at the moment we are not there.

>  By the way, what happens if we define
more drives in dosemu.conf than the default
lastdrive allows?

The lastdrive value inside DOS does not mirror the value set by `LASTDRIVE=X` in config.sys. Like I mentioned before it's built from 3 sources since DOS v3 and it's the maximum of; the number of logical drives; the value of LASTDRIVE in config.sys; 5
See [rbil](http://www.ctyme.com/intr/rb-2570.htm) so you might think you are 'setting' the value in config.sys, but in reality you can't set it less than 5 or the number of drives on the system.

Reply:
> Like I mentioned before it's built from 3 sources since DOS v3 and
> it's the maximum of; the number of logical drives; the value of LASTDRIVE in config.sys; 5

So in this case lastdrive covers all the
drives we need to redirect. Then what is
the problem? Why would you wait for
lastdrive to "settle", if even its initial value
always covers all drives?

Reply:
That the lastdrive is wrong is just an example of the CDS being in flux, so we know the CDS is going to move and at present we can't handle that.

Reply:
But we can remove lastdrive usage
completely and just use config.hdisks,
cant we?